pandas/core/nanops.py

```diff
@@ -486,6 +486,12 @@ def nanany(
     False
     """
     values, _, _, _, _ = _get_values(values, skipna, fill_value=False, mask=mask)
+
+    # For object type, any won't necessarily return
```

Comment:
can you do this in _get_values?

Reply:
Right now `_get_values` is unaware of the nanop it is being used for, so would need slight refactoring to allow that. 

While the current solution duplicates the workaround, I like it better than adding in `_get_values` since this workaround only affects any/all.  I think keeping this weird condition out of a path all other nanops hit makes sense.